pkg/kube/secretcontroller/secretcontroller.go

```diff
@@ -157,46 +154,50 @@ func NewController(
 	return controller
 }
 
+func (c *Controller) AddHandler(mc MulticlusterHandler) {
+	c.controllers = append(c.controllers, mc)
+}
+
 // Run starts the controller until it receives a message over stopCh
-func (c *Controller) Run(stopCh <-chan struct{}) {
+func (c *Controller) Run(stopCh <-chan struct{}) error {
```

Comment:
nit: why did we add an error here if we always return nil? or is this implementing some interface?

Reply:
Yep, the `kube/controller/multicluster.go` does return errors.